### DISCUSSION TITLE
Offer multiple OpenID options to user

### DISCUSSION THREAD

**Discussion by padraic-shafer** [h2oai/wave#1754] [Q&A] [answered]
From the [documentation](https://wave.h2o.ai/docs/security#single-sign-on) it looks like I can supply configuration parameters (arguments) to the wave server for **only one** OpenID authentication server.

Is there a way to let Wave know that more than one OpenID authentication option is available for the user?

For example, I would like to allow the user to authenticate via Google or ORCID.

**Comment by mturoci** [comment, marked as answer]
> it looks like I can supply configuration parameters (arguments) to the wave server for only one OpenID authentication server.

That's correct.

> Is there a way to let Wave know that more than one OpenID authentication option is available for the user?

Currently not, but could be useful for sure. You can open a feature request.

> For example, I would like to allow the user to authenticate via Google or ORCID.

Wave currently only supports OIDC (as it's the most widespread I believe), which means only Google would be possible. You can file a separate feature request for the ORCID auth.

Good suggestions @pshafer-als!

**Reply by padraic-shafer** [reply]
> > Is there a way to let Wave know that more than one OpenID authentication option is available for the user?
> 
> Currently not, but could be useful for sure. You can open a feature request.

Thanks.  I'll suggest the feature request.

I believe that [ORCID also supports OIDC](https://info.orcid.org/orcid-openid-connect-and-implicit-authentication/).

**Comment by padraic-shafer** [comment]
Feature request has been submitted: https://github.com/h2oai/wave/issues/1758

@mturoci : Thanks for your swift replies and encouragement.
